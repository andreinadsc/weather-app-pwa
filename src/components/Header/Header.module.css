.header {
    height: 10vh;
    display: flex;
    align-items: center;
    justify-content: center;
}

.logo {
    width: 8rem;
    height: 8rem;
    border-radius: 10rem;
    padding: 1rem;
}

.title {
    text-transform: uppercase;
    color: #FBFFF1;
    font-size: 3rem;
}


@media (max-width: 728px) {
    .header {
        margin-right: 5%;
    }
    .title {
        font-size: 1.5em;
    }
    .logo {
        width: 6rem;
        height: 6rem;
    }
}
