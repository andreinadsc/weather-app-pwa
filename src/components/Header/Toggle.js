import { useContext, useState } from 'react';
import classes from './Toggle.module.css';
import WeatherContext from '../../context/weather-context';

const Toggle = () => {
    const { temperatureUnitHandler } = useContext(WeatherContext);
    const [checked, setChecked] = useState(true);

    const checkedHandler = (event) => {
        temperatureUnitHandler(event.target.checked ? 'metric' : 'imperial');
        setChecked(event.target.checked);
    }

    return (
        <div className={classes['toggle-switch']}>
            <input type='checkbox' className={classes['checkbox']}
                name='temperature' id='temperature' onChange={checkedHandler} checked={checked} />
            <label className={classes['label']} htmlFor='temperature' >
                <span className={classes['inner']}  />
                <span className={classes['switch']} data-testid='switch' />
            </label>
        </div>
    );
};

export default Toggle;