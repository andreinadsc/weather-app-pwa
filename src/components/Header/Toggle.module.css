.toggle-switch {
    width: 7.5rem;
    position: absolute;
    top: 1.3%;
    right: 0;
    margin: 1rem;
    font-size: 2rem;
}

.checkbox {
    display: none;
}

.label {
    display: block;
    overflow: hidden;
    cursor: pointer;
    border: 0 solid #bbb;
    border-radius: 20px;
}

.inner {
    display: block;
    width: 200%;
    margin-left: -100%;
    transition: margin 0.3s ease-in 0s;
}

.inner:before,
.inner:after {
    float: left;
    width: 50%;
    height: 3.6rem;
    padding: 0;
    line-height: 3.6rem;
    color: var(--secondary);
    font-weight: bold;
    box-sizing: border-box;
}

.inner:before {
    content: "°C";
    padding-left: 10%;
    background-color: var(--primary);
}

.inner:after {
    content: "°F";
    padding-right: 15%;
    background-color: #bbb;
    text-align: right;
}

.switch {
    display: block;
    width: 24px;
    margin: 5px;
    background: #FBFFF1;
    position: absolute;
    top: 0;
    bottom: 0;
    right: 40px;
    border: 0 solid #bbb;
    border-radius: 20px;
    transition: all 0.3s ease-in 0s;
}

.checkbox:checked+.label .inner {
    margin-left: 0;
}

.checkbox:checked+.label .switch {
    right: 0px;
}

@media screen and (max-width: 768px) {
    .toggle-switch {
        width: 6.5rem;
        font-size: 1.5rem;
    }
}