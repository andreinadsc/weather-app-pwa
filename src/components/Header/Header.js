import Toggle from './Toggle';
import Logo from '../../images/logo.webp';
import classes from './Header.module.css';

const Header = () => {
    return (
        <header className={classes.header}>
            <Toggle />
            <img className={classes.logo} alt='logo' src={Logo} />
            <h1 className={classes.title}>Weather Forecast</h1>
        </header>
    );
};

export default Header;