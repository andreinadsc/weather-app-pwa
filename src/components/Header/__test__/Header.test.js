import { render } from '@testing-library/react'
import Header from '../Header';

describe('Header', () => {
    test('should render the Header Component correctly', () => {
        const { baseElement } = render(<Header />);

        expect(baseElement).toMatchSnapshot();
    });
});