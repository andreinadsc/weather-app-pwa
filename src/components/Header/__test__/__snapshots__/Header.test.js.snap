// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the Header Component correctly 1`] = `
<body>
  <div>
    <header
      class="header"
    >
      <div
        class="toggle-switch"
      >
        <input
          checked=""
          class="checkbox"
          id="temperature"
          name="temperature"
          type="checkbox"
        />
        <label
          class="label"
          for="temperature"
        >
          <span
            class="inner"
          />
          <span
            class="switch"
            data-testid="switch"
          />
        </label>
      </div>
      <img
        alt="logo"
        class="logo"
        src="logo.webp"
      />
      <h1
        class="title"
      >
        Weather Forecast
      </h1>
    </header>
  </div>
</body>
`;
